pkg/ipamd/ipamd.go

```diff
@@ -1197,20 +1215,74 @@ func (c *IPAMContext) logPoolStats(dataStoreStats *datastore.DataStoreStats) {
 	log.Debugf("%s: %s, c.maxIPsPerENI = %d", prefix, dataStoreStats, c.maxIPsPerENI)
 }
 
-func (c *IPAMContext) askForTrunkENIIfNeeded(ctx context.Context) {
-	if c.enablePodENI && c.dataStore.GetTrunkENI() == "" {
+func (c *IPAMContext) askForTrunkENIIfNeeded(ctx context.Context, labels map[string]string) error {
+	var err error
+	hasTrunkLabel := false
```

Comment:
I see here if labels are `nil`, hasTrunkLabel is false so we end up requesting a trunk ENI, how is it handled in VPC RC if trunk is attached and for whatever reason node label is missing? Do we just retry adding the label?

Reply:
Yes, I am adding a check in resource controller to handle this. I don't think we handled the nil map in labels in current codes. I am thinking to either retry in resource controller in case API server error, or add a map to pod as labels and add the label.